speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/FileDataStorageTests.cpp

```diff
@@ -399,4 +433,275 @@ TEST(FileDataStorageTests, WriteSameFileTwice)
+        int compressionRate = random.NextInt(1, COMPRESSION_LIMIT / 2) * 2 - 1;
+        fileDataStorage.SetCompressionRate(compressionRate);
+
+        float value = random.NextFloat();
+        auto compressionStep = static_cast<unsigned long>(random.NextUInt(compressionRate));
+
+        float values[COMPRESSION_LIMIT];
+        float originalValues[COMPRESSION_LIMIT];
+        for (int j = 0; j < compressionRate; j++)
+        {
+            float randomValue = random.NextFloat();
+            values[j] = randomValue;
+            originalValues[j] = randomValue;
+        }
+
+        int placeInArray = static_cast<int>(compressionStep % static_cast<unsigned long>(compressionRate));
+
+        fileDataStorage.AddToArray(values, value, compressionStep);
+
+        for (int j = 0; j < compressionRate; j++)
+        {
+            if (j == placeInArray)
+            {
+                ASSERT_EQ(values[placeInArray], value);
+            }
+            else
+            {
+                ASSERT_EQ(values[j], originalValues[j]);
+            }
+        }
+    }
+}
+
+/// @brief Test whether GetMedian returns the median of a given array
+TEST(FileDataStorageTests, GetMedianTest)
+{
+    Random random;
+
+    for (int i = 0; i < TEST_AMOUNT; i++)
+    {
+        float values[COMPRESSION_LIMIT];
+        float check[COMPRESSION_LIMIT];
+
+        // compression rate uneven and > 0
+        int compressionRate = random.NextInt(1, COMPRESSION_LIMIT / 2) * 2 - 1;
+
+        FileDataStorage fileDataStorage;
+        fileDataStorage.SetCompressionRate(compressionRate);
+
+        for (int j = 0; j < compressionRate; j++)
+        {
+            float rnd = random.NextFloat(0, 100);
+            values[j] = rnd;
+            check[j] = rnd;
+        }
+
+        float testMedian = fileDataStorage.GetMedian(values);
+
+        ASSERT_EQ(HelperGetMedian(check, compressionRate), testMedian);
+    }
+}
+
+/// @brief Test whether GetLeastCommon returns the least common value of a given array
+TEST(FileDataStorageTests, GetLeastCommonTest)
+{
+    Random random;
+    FileDataStorage fileDataStorage;
+    fileDataStorage.SetCompressionRate(COMPRESSION_RATE);
+
+    for (int i = 0; i < TEST_AMOUNT; i++)
+    {
+        int values[COMPRESSION_RATE];
+        int leastCommonVal = random.NextInt();
+        int leastCommonFrequency = random.NextInt(1, COMPRESSION_RATE);
+
+        if (leastCommonFrequency > COMPRESSION_RATE / 2)
+        {
+            for (int j = 0; j < COMPRESSION_RATE; j++)
+            {
+                values[j] = leastCommonVal;
+            }
+        }
+        else
+        {
+            for (int j = 0; j < COMPRESSION_RATE; j++)
+            {
+                if (j < leastCommonFrequency)
+                {
+                    values[j] = leastCommonVal;
+                }
+                else
+                {
+                    values[j] = leastCommonVal + 1;
+                }
+            }
+        }
```

Comment:
too much indentation here, all elses should be removed and replaced by continue, inverting or extracting a function